Sources/ProjectDescription/InfoPlist.swift

```diff
@@ -2,6 +2,7 @@ import Foundation
 
 public enum InfoPlist: Codable {
     case file(path: String)
+    case dictionary([String: Any])
```

Comment:
What types can be inside of the dictionary? Is it worth being stricter here?

```swift
enum InfoPlist {

    typealias Key = String
    
    enum Value {
        case string(String)
        case integer(Int)
    }
    
    case dictionary([Key: Value])
    
}

extension InfoPlist.Value: ExpressibleByStringLiteral {
    init(stringLiteral value: String) {
        self = .string(value)
    }
}

extension InfoPlist.Value: ExpressibleByIntegerLiteral {
    init(integerLiteral value: Int) {
        self = .integer(value)
    }
}

InfoPlist.dictionary([ "NSPrincipleClass": "$(PRODUCT_NAME).TuistTestObserver" ])
```

We could also introduce Key as an enum, but that may be overkill.

```swift
    enum Key {
        case NSPrincipleClass
        case UILaunchImages
        case custom(String)
    }
```

Reply:
@ollieatkinson I've pushed a commit with this suggestion. I think it's a great idea. For now, I defined the `Key` as a string but in the future we can do what you suggested, having an enum that contains the most common keys together with the documentation.

Reply:
Looks great, and removes the need for the workaround which is even better! I agree about Key being a string - keeps it simple for now.